presto-cassandra/src/main/java/com/facebook/presto/cassandra/CassandraTokenSplitManager.java

```diff
@@ -116,8 +118,11 @@ public CassandraTokenSplitManager(CassandraSession session, int splitSize)
         return result.build();
     }
 
-    private long getTotalPartitionsCount(String keyspace, String table)
+    public long getTotalPartitionsCount(String keyspace, String table)
     {
+        if (splitsPerNode != null) {
+            return splitsPerNode;
```

Comment:
can there be fewer partitions than the configured value (like empty data source) ?